src/DsaGlobal.cc

```diff
@@ -309,6 +311,11 @@ bool ContextSensitiveGlobalAnalysis::runOnModule(Module &M) {
   BottomUpAnalysis bu(m_dl, m_tli, m_allocInfo, m_cg, flowSensitiveOpt);
   bu.runOnModule(M, m_graphs);
 
+  // cloning the bu graphs for later
```

Comment:
this seems to be unconditional. It might be very expensive on a large program. 
Only save graphs if they will actually be used later.

Reply:
I agree. What would be the correct way to decide to clone?
Currently, the pass is run if the option `MemDot` is detected (in `tools/seadsa.cc`).

Reply:
We have several examples of global variables shared across different files

Reply:
It is better to pass a flag to the pass (maybe during creation time) instead of propagating dependence on global variables.

Reply:
I think that the `ContextSensitiveGlobalPass` is created with
`AU.addRequired<ContextSensitiveGlobalPass>();` in `getAnalysisUsage`, and by calling the default constructor. Do you know if there is a way to add the flag there?

Reply:
If we are using LLVM pass manager to instantiate things for us, then there is no way to pass arguments. Relying on the pass manager was an unfortunate design decision. It is better to call the necessary code directly from inside the path. But at this point, it might be easier for you to make the global variable accessible where you need it.